.github/workflows/checks.yml

```diff
@@ -11,8 +11,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v5
         with:
-          go-version: "1.21"
-          check-latest: true
+          go-version-file: go.mod
```

Comment:
Latest (v1.21.10) fails testing on macOS (macOS's new linker doesn't work, this is only a problem with the `-race` flag, which requires CGO)